### PR TITLE
[Enhancement] Improve stream load rpc timeout and retry

### PR DESCRIPTION
**Pull request by wyb**
Why I'm doing:

If it takes a long time to acquire the db lock, stream load put and commit rpc will time out and report an error `call frontend service failed` to the user.

What I'm doing:
1. use db try lock with rpc_timeout when db lock is hold by others, more clear error information will be returned to the user and the lock information can be writed into the log in time.
2. add stream load put retry.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by meegoo — DISMISSED** [review]

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195434678。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195434678.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195435882。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195435882.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195436949。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195436949.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195438055。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195438055.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195439134。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195439134.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195440162。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195440162.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195441279。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195441279.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195442549。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195442549.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195443661。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195443661.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195445006。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195445006.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195446196。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195446196.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195447514。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195447514.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195448880。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195448880.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195450038。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195450038.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195451446。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195451446.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195452677。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195452677.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195453775。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195453775.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195454884。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195454884.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195456184。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195456184.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195457230。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195457230.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195458620。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195458620.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195459405。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195459405.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195460525。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195460525.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195461884。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195461884.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195463005。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195463005.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195464692。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195464692.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195465752。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195465752.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195466907。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195466907.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

**Comment by llkj1** [comment]
感谢您与我们联系。(Thank you for contacting us.)
------------------------------------------------------------

我们收到了您的请求，并会在一至两个工作日内给您回复。您的案例编号是 102195468084。

想了解更多与开发相关主题的信息，请访问 开发者支持网页 (https://developer.apple.com/support/)。
                  
Apple Developer Program Support


We’ve received your support request and will get back to you in one to two business days. Your case number is 102195468084.

For additional information on development-related topics, visit:
https://developer.apple.com/support/

Best regards, 

Apple Developer Program Support
           

------------------------------------------------------------
Copyright (c) 2024 Apple Inc. All rights reserved.

Contact Us
https://developer.apple.com/contact/

Developer
https://developer.apple.com/

My Apple ID
https://appleid.apple.com

Privacy Policy
https://www.apple.com/privacy/

File: be/src/runtime/stream_load/stream_load_executor.cpp [hunk 1]
```diff
@@ -223,54 +224,50 @@ Status StreamLoadExecutor::commit_txn(StreamLoadContext* ctx) {
...
-    if (st.is_service_unavailable()) {
-        auto remain_timeout_ms = ctx->load_deadline_sec > 0 ? (ctx->load_deadline_sec - UnixSeconds()) * 1000 : 0;
-        if (remain_timeout_ms > 0) {
-            return commit_txn_internal(request, remain_timeout_ms, ctx);
+    int retry = 0;
+    TLoadTxnCommitResult result;
+    while (true) {
+        RETURN_IF_ERROR(commit_txn_internal(request, rpc_timeout_ms, &result));
+        Status st(result.status);
+        if (st.ok()) {
```
Comment (banmoy) [inline be/src/runtime/stream_load/stream_load_executor.cpp:232]: what about `st.is_service_unavailable()`

**Comment by banmoy** [comment]
also need to deal with `StreamLoadExecutor::prepare_txn`

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 2]
```diff
@@ -541,9 +544,11 @@ static void throttleCommitOnRateExceed(CommitRateExceededException e, long start
     private VisibleStateWaiter commitTransactionUnderDatabaseWLock(
             @NotNull Database db, long transactionId, @NotNull List<TabletCommitInfo> tabletCommitInfos,
             @NotNull List<TabletFailInfo> tabletFailInfos,
-            @Nullable TxnCommitAttachment attachment) throws UserException {
+            @Nullable TxnCommitAttachment attachment, long timeoutMs) throws UserException {
         Locker locker = new Locker();
-        locker.lockDatabase(db, LockType.WRITE);
+        if (!locker.tryLockDatabase(db, LockType.WRITE, timeoutMs)) {
+            throw new LockTimeoutException("get database write lock timeout, database=" + db.getFullName());
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java]: done

File: be/src/runtime/stream_load/stream_load_executor.cpp (hunk 1 above)
Comment (wyb) [reply be/src/runtime/stream_load/stream_load_executor.cpp:232]: there is no service unavailable error now. 
when encountering `TTransportException`, rpc will try again in `thrift_rpc_helper`.

**Review by banmoy — APPROVED** [review]

**Review by sevev — APPROVED** [review]

**Review by mervinsr — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by kevincai — APPROVED** [review]

**Comment by wyb** [comment]
@Mergifyio backport branch-3.2

**Pull request merged**